src/main/kotlin/racing/view/CarRacingResultView.kt

```diff
@@ -0,0 +1,32 @@
+package racing.view
+
+import racing.dto.Car
+import racing.dto.Winners
+
+object CarRacingResultView {
+    fun printCarRacingResult(cars: List<Car>, moveCount: Int) {
+        // turnOfPosition의 0번 index는 기본으로 1이 셋팅되어있기때문에
+        // 범위를 0 <= turn <= moveCount 로 설정
+        for (turn in 0..moveCount) {
+            printMovingDistanceByTurn(cars = cars, turn = turn)
+        }
+    }
+
+    private fun printMovingDistanceByTurn(cars: List<Car>, turn: Int) {
```

Comment:
```suggestion
    private inline fun printMovingDistanceByTurn(cars: List<Car>, positionOf: (Car) -> Int) {
```
변수의 전달범위를 조절하면 가독성이 올라가기도 합니다. 다만 함수형 프로그래밍에 익숙하지 않다면 오히려 방해가 되니 적절히 사용해야 해요. 브랜은 어떠신가요.

관련 피드백: https://github.com/next-step/kotlin-racingcar/pull/526#discussion_r872550955

Reply:
해당 부분이 익숙한 모습이 아니어서 그런 것인지.. 
어떤 점이 가독성을 올리는 것인지 이해가 어렵습니다ㅜㅜ..

```
private fun printMovingDistanceByTurn(cars: List<Car>, turn: Int) {
    cars.forEach { car ->
        printCurrentCarMovingDistance(carName = car.name, carPosition = car.turnOfPosition[turn])
    }
    println()
}
```

car의 속해있는 정보들을 이용해서 printCurrentCarMovingDistance() 함수를 호출하고 있습니다.
position도 name과 같은 방식으로 파라미터로 전달하는 편이 같은 객체내의 정보를 전달한다고 직관적으로 표현하는 것이 아닐까 생각이 드네요..!

의견 주시면 감사하겠습니다!!

Reply:
계속 고민하다보니 결국에는 car 객체를 넘겨서 차이가 없군요..!!
제가 질문하면서 글을 적다가 정리가 되었습니다

```
private inline fun printMovingDistanceByTurn(cars: List<Car>, positionOf: (Car) -> Int) {
    cars.forEach { car ->
        printCurrentCarMovingDistance(carName = car.name, carPosition = positionOf(car))
    }
    println()
}
```

Reply:
네 ㅎㅎ https://github.com/next-step/kotlin-racingcar/pull/580#discussion_r873574071 에서도 말씀드린것처럼 조직내 숙련도에 따라 더 안좋을 수도 있어요. for vs forEach 같이요